### ISSUE TITLE
Multi-user support

### ISSUE THREAD

**Issue by Rillke** [hwdsl2/docker-ipsec-vpn-server#48]
In order to support multiple accounts, I had to customize the `CMD` (run.sh) script either to allow the specification of multiple users **or** adding a check whether there are accounts in `/etc/ppp/chap-secrets` and `/etc/ipsec.d/passwd` before overwriting them so I could mount them as volumes and fill them with users using something similar to

```
docker-compose exec vpn bash
VPN_USER='new_user'
VPN_PASSWORD='new_passwd'

cat >> /etc/ppp/chap-secrets <<EOF
"$VPN_USER" l2tpd "$VPN_PASSWORD" *
EOF

VPN_PASSWORD_ENC=$(openssl passwd -1 "$VPN_PASSWORD")
cat >> /etc/ipsec.d/passwd <<EOF
$VPN_USER:$VPN_PASSWORD_ENC:xauth-psk
EOF
```

Is not supporting multiple users a design decision or would merge requests be accepted if they do not complicate things for a default setup.

**Comment by GabLeRoux** [comment]
After reading the documentation a bit more, It doesn't seem possible to have multi user on first run (yet), but there is some documentation in the [important notes](https://github.com/hwdsl2/docker-ipsec-vpn-server#important-notes):

> If you wish to add, edit or remove VPN user accounts, see [Manage VPN Users](https://github.com/hwdsl2/setup-ipsec-vpn/blob/70c6d6b/docs/manage-users.md). Important: After editing the VPN config files, you must also comment out the relevant sections in `/opt/src/run.sh`, to avoid losing your changes on container restart.

## avoid losing changes on container restart

I suppose the author is talking about these lines:
* https://github.com/hwdsl2/docker-ipsec-vpn-server/blob/38deb861701754e2f345f7f98c176c77eb9b49b2/run.sh#L164-L167
* https://github.com/hwdsl2/docker-ipsec-vpn-server/blob/38deb861701754e2f345f7f98c176c77eb9b49b2/run.sh#L202-L205
* https://github.com/hwdsl2/docker-ipsec-vpn-server/blob/38deb861701754e2f345f7f98c176c77eb9b49b2/run.sh#L207-L210

## restart services

I suppose the only missing things from your script would be to restart the services once configurations are updated

```bash
docker-compose exec vpn service ipsec restart
docker-compose exec vpn service xl2tpd restart
```

**Comment by Rillke** [comment]
> I suppose the only missing things from your script would be to restart the services once configurations are updated

Which is basically the same as restarting the entire container.
1. Is restarting the services really necessary (`service ipsec restart` [does the same as `reload`](https://github.com/libreswan/libreswan/blob/master/initsystems/sysvinit/ipsec.init.in#L282))?
2. Have you found a disruption-free way adding new users?

**Comment by GabLeRoux** [comment]
> 1. Is restarting the services really necessary (service ipsec restart does the same as reload)?

I did not try it yet, but I if I remember well, `reload` for `nginx` reloads configuration without killing the process and will keep running with older configuration in case of a wrong/broken configuration, preventing the service to go down, which is a good thing. So reloading is probably the right thing to do here instead.

> connections are closed when reloading the ipsec daemon

💩 that is not cool and I did not find anything online concerning this.
